.pipelines/templates/ToolchainBuild.yml

```diff
@@ -54,3 +54,31 @@ steps:
       tar -C "${{ parameters.buildRepoRoot }}/build/logs/toolchain" -czf "$published_logs_dir/toolchain.logs.tar.gz" .
     condition: always()
     displayName: "Copy logs for publishing"
+
+  - bash: |
+      # Publish the list of packages that should be re-tested during full package build.
+      #   This list will be the contents of 'built_specs_list.txt' in the toolchain build logs directory, but only for
+      #   packages that have a '%check' section in their spec file. The assumption is that all packages will have a 
+      #   '%check' section for all architectures, or none.
+      built_list="${{ parameters.buildRepoRoot }}"/build/logs/toolchain/built_specs_list.txt
+      specs_dir="${{ parameters.buildRepoRoot }}/SPECS/"
+      if [[ -f "$built_list" ]]; then
+        built_specs=$(xargs < "$built_list")
+        retest_list=()
+        for s in ${built_specs}; do
+          if [ ! -f "${specs_dir}/${s}/${s}".spec ]; then
+              echo "ERROR: '${specs_dir}/${s}/${s}.spec' does not exist"
+              exit 1
+          fi
+          if grep -q '^%check' "${specs_dir}/${s}/${s}.spec"; then
+              retest_list+=("${s}")
+          fi
+        done
+      else
+        echo "No file '$built_list' found, so no packages to re-test"
+      fi
+      # Default will be "", which is fine.
+      echo "Setting 'CalculatePackageRetestList.packageRetestList' to '${retest_list[*]}'"
+      echo "##vso[task.setvariable variable=packageRetestList;isOutput=true]${retest_list[*]}"
+    name: "CalculatePackageRetestList"
+    displayName: "Calculating packages that should be re-tested"
```

Comment:
The logic looks good. I want to keep the template YAMLs dumb and focused only on their own small tasks without knowledge of any additional tasks the user might want to perform like re-building the toolchain package for the sake of testing or the `SPECS` directory. With that in mind, could we move this logic into `PackageBuildPRCheck.yml`? An additional bonus here would be that we're setting and using the job variable `packageRetestList` in the same file, so it's easier to track where it came from.

Reply:
Hmm, might be interesting in regards to handling the artifacts. Do we want a new artifact, or just add to the existing toolchain artifact?